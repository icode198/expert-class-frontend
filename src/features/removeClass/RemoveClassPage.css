.remove-class-container {
  width: 1050px;
  height: 100vh;
  margin-left: 250px;
  padding: 10px;
}

th,
td {
  padding: 10px;
  text-align: left;
  border: 1px solid #333;
}

th {
  background-color: #97bf0f;
}

table tr:nth-child(odd) {
  background-color: #eee;
}