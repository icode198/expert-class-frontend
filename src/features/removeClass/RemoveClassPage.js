import React from 'react';
// import { useSelector } from 'react-redux';
import './RemoveClassPage.css';

function RemoveClassPage() {
  // const { classes } = useSelector((state) => state.classes);
  const classes = [
    {
      id: 1,
      title: 'Setup Redux',
      description: 'Learn how to set up Redux on a React app.',
      instructor: 'Breno',
      duration: 4,
      image: 'image-url',
    },
    {
      id: 2,
      title: 'Setup Redux',
      description: 'Learn how to set up Redux on a React app.',
      instructor: 'Breno',
      duration: 4,
      image: 'image-url',
    },
    {
      id: 3,
      title: 'Setup Redux',
      description: 'Learn how to set up Redux on a React app.',
      instructor: 'Breno',
      duration: 4,
      image: 'image-url',
    },
  ];

  const handleRemove = (id) => {
    fetch(`.../api/v1/remove_class/${id}`);
  };

  const classesList = classes.map((obj) => (
    <div key={obj.id} className="class_item">
      <h3>{obj.title}</h3>
      <h4>{obj.description}</h4>
      <p>{obj.instructor}</p>
      <p>{obj.duration}</p>
      <p>{obj.image}</p>

      <button type="button" className="remove_btn" onClick={handleRemove(obj.class_id)}>Remove</button>
    </div>
  ));

  return (
    <div>
      <h1>RemoveClassPage</h1>
      {classesList}
    </div>
  );
}

export default RemoveClassPage;
