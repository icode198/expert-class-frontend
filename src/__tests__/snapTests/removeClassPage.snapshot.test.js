import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithRedux } from '../../tests/test-utils';
import RemoveClassPage from '../../pages/removeClassPage/removeClassPage';

afterEach(cleanup);

describe('RemoveClassPage', () => {
  test('Renders the page', async () => {
    const app = renderWithRedux(<RemoveClassPage />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
