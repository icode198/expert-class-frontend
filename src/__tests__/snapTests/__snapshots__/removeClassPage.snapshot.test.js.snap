// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveClassPage Renders the page 1`] = `
<div
  className="remove-class-container"
>
  <div
    className="heading"
  >
    Remove Class
  </div>
  <table
    className="reservations-table"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Description
        </th>
        <th>
          Instructor
        </th>
        <th>
          Duration
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
